ci/scripts/find-regression.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+
+import subprocess
+import unittest
+import os
+import sys
+import time
+
+'''
+@kwannoel
+This script is used to find the commit that introduced a regression in the codebase.
+It uses binary search to find the regressed commit.
+It works as follows:
+1. Use the start (inclusive) and end (exclusive) bounds, find the middle commit.
+   e.g. given commit 0->1(start)->2->3->4(bad), start will be 1, end will be 4. Then the middle commit is (1+4)//2 = 2
+        given commit 0->1(start)->2->3(bad)->4, start will be 1, end will be 3. Then the middle commit is (1+3)//2 = 2
+        given commit 0->1(start)->2(bad), start will be 1, end will be 2. Then the middle commit is (1+2)//2 = 1.
+        given commit 0->1(start,bad), start will be 1, end will be 1. We just return the bad commit (1) immediately.
```

Comment:
Shall we update the doc, since we're now calling `git bisect` instead of running the algorithm ourselves?

Reply:
I'm not calling git bisect anymore.